tls_sw.c

```diff
@@ -1939,6 +1942,30 @@ int tls_sw_recvmsg(struct sock *sk,
 
 	copied += decrypted;
 
+	plaintext = kmalloc(copied, GFP_KERNEL);
+	err = copy_from_user(plaintext, msg->msg_iter.iov->iov_base, copied);
+	if (err) {
+		kfree(plaintext);
+		copied = 0;
+		err = -EAGAIN;
+		tls_ctx->lua_err = TLS_LUA_RECVERR;
+		goto end;
+	}
+	if (tls_ctx->L && tls_ctx->recv_entry) {
+		lua_getglobal(tls_ctx->L, tls_ctx->recv_entry);
+		lua_pushlstring(tls_ctx->L, plaintext, copied);
+		lua_pushnumber(tls_ctx->L, copied);
```

Comment:
When you push the string above, Lua already knows the payload length. No need to push it here.